src/lib/Libifl/tm.c

```diff
@@ -711,11 +712,20 @@ tm_attach(char *jobid, char *cookie, pid_t pid, tm_task_id *tid, char *host, int
 #endif
 
 	pbs_tcp_interrupt = 1;
-	tm_jobid = jobid;
-	tm_jobid_len = (tm_jobid != NULL) ? strlen(tm_jobid) : 0;
 
-	tm_jobcookie = cookie;
-	tm_jobcookie_len = (tm_jobcookie != NULL) ? strlen(tm_jobcookie) : 0;
+	tm_jobid_len = 0;
+	free(tm_jobid);
+	tm_jobid = strdup(jobid);
+	if (!tm_jobid)
+		return TM_ESYSTEM;
+	tm_jobid_len = strlen(tm_jobid);
+
+	tm_jobcookie_len = 0;
+	free(tm_jobcookie);
+	tm_jobcookie = strdup(jobid);
```

Comment:
i think this is supposed to be tm_jobcookie = cookie.

Reply:
That may have been the case at some point in time, but I don't believe it still is. If you track down all the calls to tm_attach() they emanate from the pbs_attach command. I don't see anywhere where the cookie passed is anything other than NULL. Did I miss something?

Reply:
Originally, it was "tm_jobcookie = cookie;" under tm_attach(), and now "tm_jobcookie = strdup(jobid);" and jobid is not NULL. So perhaps it was really expecting it to be NULL?

Reply:
I guess if cookie is NULL, then need to pass it as is:

08/27/2018 19:31:40  S    Obit received momhop:1 serverhop:1 state:4 substate:42
**08/27/2018 19:31:40  M    The operation completed successfully.tm_request, tm_attach: job cookie is not NULL**
08/27/2018 19:31:40  M    task 1 terminated
08/27/2018 19:31:40  M    Terminated

Reply:
I don't want to change the code when I don't know how it will affect the behavior, so I would prefer to leave it alone.

Reply:
I see your point @bayucan. Code has been updated.